internal/handler/wallet/import.go

```diff
@@ -1,99 +1,51 @@
 package wallet
 
 import (
-	"github.com/go-openapi/runtime/middleware"
-	"lukechampine.com/blake3"
+	"sync"
 
+	"fyne.io/fyne/v2"
+	"github.com/go-openapi/runtime/middleware"
 	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/models"
 	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi/operations"
-
-	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/password"
 	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/wallet"
 )
 
-// checkDuplicate checks that the same wallet doesn't already exist.
-func checkDuplicate(nickname string) middleware.Responder {
-	_, err := wallet.Load(nickname)
-	if err == nil {
-		return operations.NewRestWalletImportInternalServerError().WithPayload(
-			&models.Error{
-				Code:    errorAlreadyExists,
-				Message: "Error: a wallet with the same nickname already exists.",
-			})
-	}
-
-	return nil
+//nolint:nolintlint,ireturn
+func NewImport(walletStorage *sync.Map, app *fyne.App) operations.RestWalletImportHandler {
```

Comment:
walletStorage should be removed. fyne.App should be replaced with a proper inteface